zerver/webhooks/groove/view.py

```diff
@@ -0,0 +1,102 @@
+        body = u"An open ticket has been assigned to"
+    else:
+        body = u"A {state} ticket has been assigned to"
+
+    # If there is a person and/or a group assigned,
+    # make a notification message. Otherwise, ignore it.
+    if assignee or assigned_group:
+        if assignee and assigned_group:
+            body += u" {assignee} from {assigned_group}"
+        elif assignee:
+            body += u" {assignee}"
+        elif assigned_group:
+            body += u" {assigned_group}"
+        body += u"\n```quote\n**[Ticket #{number}: {title}]({app_url})**\n```"
+        return body.format(**payload)
+    else:
+        return None
+
+def agent_replied_body(payload: Dict[str, Any]) -> Text:
+    # Take the agent's email and the ticket number from the payload.
+    agent = payload['links']['author']['href'].split("http://api.groovehq.com/v1/agents/")[1]
+    number = payload['links']['ticket']['href'].split("http://api.groovehq.com/v1/tickets/")[1]
+
+    # Create the notification message.
+    body = u"%s has just replied to a ticket\n```quote\n**[Ticket #%s]" % (agent, number)
+    body += u"({app_ticket_url})**\n{plain_text_body}\n```"
+    return body.format(**payload)
+
+def customer_replied_body(payload: Dict[str, Any]) -> Text:
+    # Take the customer's email and the ticket number from the payload.
+    customer = payload['links']['author']['href'].split("http://api.groovehq.com/v1/customers/")[1]
+    number = payload['links']['ticket']['href'].split("http://api.groovehq.com/v1/tickets/")[1]
+
+    # Create the notification message.
+    body = u"%s has just replied to a ticket\n```quote\n**[Ticket #%s]" % (customer, number)
+    body += u"({app_ticket_url})**\n{plain_text_body}\n```"
+    return body.format(**payload)
+
+def note_added_body(payload: Dict[str, Any]) -> Text:
+    # Take the agent's email and the ticket number from the payload.
+    agent = payload['links']['author']['href'].split("http://api.groovehq.com/v1/agents/")[1]
+    number = payload['links']['ticket']['href'].split("http://api.groovehq.com/v1/tickets/")[1]
+
+    # Create the notification message.
+    body = u"%s has left a note\n```quote\n**[Ticket #%s]" % (agent, number)
+    body += u"({app_ticket_url})**\n{plain_text_body}\n```"
+    return body.format(**payload)
+
+@api_key_only_webhook_view('Groove')
+@has_request_variables
+def api_groove_webhook(request: HttpRequest, user_profile: UserProfile,
+                       payload: Dict[str, Any]=REQ(argument_type='body'),
+                       stream: Text=REQ(default='groove'),
+                       topic: Optional[Text]=REQ(default='notifications')) -> HttpResponse:
+    # The event identifier is stored in the X_GROOVE_EVENT header.
+    event = request.META['X_GROOVE_EVENT']
+
+    # We listen to several events that are used for notifications.
+    # Other events are ignored.
+    if event in EVENTS_FUNCTION_MAPPER:
+        body = EVENTS_FUNCTION_MAPPER[event](payload)
+    else:
```

Comment:
It looks like you could simplify this code by removing the else clause and indenting the following "if body is not None:" check.

Reply:
Fixed!